pkg/cmd/roachtest/tests/failover.go

```diff
@@ -70,6 +77,74 @@ func registerFailover(r registry.Registry) {
 	}
 }
 
+// 5 nodes fully connected. Break the connection between a pair of nodes 4 and 5
+// while running a workload against nodes 1 through 3. Before each disconnect,
+// move all the leases to nodes 4 and 5 in a different pattern.
+func runDisconnect(ctx context.Context, t test.Test, c cluster.Cluster) {
+	require.Equal(t, 6, c.Spec().NodeCount)
+	require.False(t, c.IsLocal(), "test can't use local cluster") // messes with iptables
+
+	// Create cluster.
+	opts := option.DefaultStartOpts()
+	settings := install.MakeClusterSettings()
+
+	c.Put(ctx, t.Cockroach(), "./cockroach")
+	c.Start(ctx, t.L(), opts, settings, c.Range(1, 5))
+
+	conn := c.Conn(ctx, t.L(), 1)
+	defer conn.Close()
+
+	// Wait for upreplication.
+	require.NoError(t, WaitFor3XReplication(ctx, t, conn))
+
+	t.Status("creating workload database")
+	_, err := conn.ExecContext(ctx, `CREATE DATABASE kv`)
+	require.NoError(t, err)
+
+	c.Run(ctx, c.Node(6), `./cockroach workload init kv --splits 100 {pgurl:1}`)
+
+	// Start workload on n6 using nodes 1-3 (not part of partition). We could
+	// additionally test the behavior of running SQL against nodes 4-5 however
+	// that complicates the analysis as we want to focus on KV behavior.
+	t.Status("running workload")
+	m := c.NewMonitor(ctx, c.Range(1, 3))
+	m.Go(func(ctx context.Context) error {
+		c.Run(ctx, c.Node(6), `./cockroach workload run kv --read-percent 50 `+
+			`--duration 10m --concurrency 256 --max-rate 2048 --timeout 1m --tolerate-errors `+
+			`--histograms=`+t.PerfArtifactsDir()+`/stats.json `+
+			`{pgurl:1-3}`)
+		return nil
+	})
+
+	// Start and stop partial between nodes 4 and 5 every 30 seconds.
+	m.Go(func(ctx context.Context) error {
+		// Try different combinations of the ranges between the partitioned nodes.
+		// All the system ranges will be on all the nodes, so they will all move,
+		// plus many of the non-system ranges.
+		for i := 2; i <= 7; i++ {
+			time.Sleep(30 * time.Second)
```

Comment:
Let's use 1 minute intervals here. I found that several failure modes can take more than 30 seconds to resolve, and we want to capture these in the benchmark results as well.

Let's also use a similar timer approach as in e.g. `runFailoverNonSystem`. Tickers instead of sleeps ensure that the time taken to e.g. relocate leases or ranges doesn't skew the overall runtime, which could otherwise cause the workload to stop before we're done with partitions. And the random wait ensures we start the partition at different offsets into the lease interval, since a fixed periodicity will otherwise skew the results towards artificially low pMax latencies.